### ISSUE TITLE
New Chat - ChatCraft

### ISSUE THREAD

**Issue by irthomasthomas** [irthomasthomas/undecidability#656]
- [ ] [New Chat - ChatCraft](https://chatcraft.org/c/zIpIpIKuJ8E3S8jlMsJTO)

# New Chat - ChatCraft

**DESCRIPTION:**  
Welcome to ChatCraft, your open-source web companion for coding with Large Language Models (LLMs). Designed with developers in mind, ChatCraft transforms the way you interact with GPT models, making it effortless to read, write, debug, and enhance your code.

We think ChatCraft is the best platform for learning, experimenting, and getting creative with code. Here's a few of the reasons why we think you'll agree:

| Feature                                | ChatCraft | ChatGPT | Copilot |
|----------------------------------------|-----------|---------|---------|
| Optimized for conversations about code | ✅        | ❌      | ❌      |
| Work with models from multiple AI vendors | ✅     | ❌      | ❌      |
| Previews for Mermaid Diagrams, HTML    | ✅        | ❌      | ❌      |
| Edit Generated AI Replies              | ✅        | ❌      | ✅      |
| Use Custom System Prompts              | ✅        | ✅      | ❌      |
| Easy to retry with different AI models | ✅       | ❌      | ❌      |
| Edit/Run Generated Code and Custom Functions | ✅ | ❌      | ✅      |
| Open Source                            | ✅        | ❌      | ❌      |

Learn more about ChatCraft on [GitHub](https://chatcraft.org/c/zIpIpIKuJ8E3S8jlMsJTO)

**Quick Start Instructions**  
You can begin using ChatCraft today by following these steps:

1. Choose an AI provider below: we support both OpenAI and OpenRouter. OpenAI supports various versions of ChatGPT (gpt-3.5-turbo) and GPT-4 models, while OpenRouter adds support for even more models from vendors like Anthropic, Google, and Meta. It's easy to switch providers later, or go back-and-forth.
2. Enter an API Key. ChatCraft is a "bring your own API Key" web app. No matter which provider you choose, ChatCraft needs an API Key to start making API calls on your behalf. API Keys are never shared, and get stored in your browser's local storage.
3. Start chatting with AI! Type your question in the textbox at the bottom of the screen and click the Ask button to prompt a particular model (switch to a different model whenever you like).
4. Copy, edit, delete, or retry any AI response with a different model until you get the results you need.
5. Every chat is saved to a local, offline database in your browser, which you can search (top of UI) or navigate by opening the sidebar with the hamburger menu in the top-left.

####

**Comment by irthomasthomas** [comment]
### Related issues

### #418: openchat/openchat-3.5-1210 · Hugging Face
<details><summary>### Details</summary>Similarity score: 0.91
- [ ] [openchat/openchat-3.5-1210 · Hugging Face](https://huggingface.co/openchat/openchat-3.5-1210#conversation-templates)

#### Using the OpenChat Model

We highly recommend installing the OpenChat package and using the OpenChat OpenAI-compatible API server for an optimal experience. The server is optimized for high-throughput deployment using vLLM and can run on a consumer GPU with 24GB RAM.

- **Installation Guide**: Follow the installation guide in our repository.
- **Serving**: Use the OpenChat OpenAI-compatible API server by running the serving command from the table below. To enable tensor parallelism, append `--tensor-parallel-size N` to the serving command.

    Model | Size | Context | Weights | Serving
    ------|------|---------|---------|--------
    OpenChat 3.5 | 1210 | 7B | 8192 | `python -m ochat.serving.openai_api_server --model openchat/openchat-3.5-1210 --engine-use-ray --worker-use-ray`

- **API Usage**: Once started, the server listens at `localhost:18888` for requests and is compatible with the OpenAI ChatCompletion API specifications. Here's an example request:

    ```bash
    curl http://localhost:18888/v1/chat/completions \
      -H "Content-Type: application/json" \
      -d '{
            "model": "openchat_3.5",
            "messages": [{"role": "user", "content": "You are a large language model named OpenChat. Write a poem to describe yourself"}]
          }'
    ```
- **Web UI**: Use the OpenChat Web UI for a user-friendly experience.

##### Online Deployment

If you want to deploy the server as an online service, use the following options:

- `--api-keys sk-KEY1 sk-KEY2 ...` to specify allowed API keys
- `--disable-log-requests --disable-log-stats --log-file openchat.log` for logging only to a file.

For security purposes, we recommend using an HTTPS gateway in front of the server.

##### Mathematical Reasoning Mode

The OpenChat model also supports mathematical reasoning mode. To use this mode, include `condition: "Math Correct"` in your request.

    ```bash
    curl http://localhost:18888/v1/chat/completions \
      -H "Content-Type: application/json" \
      -d '{
            "model": "openchat_3.5",
            "condition": "Math Correct",
            "messages": [{"role": "user", "content": "10.3 − 7988.8133 = "}]
          }'
    ```

##### Conversation Templates

We provide several pre-built conversation templates to help you get started.

- **Default Mode (GPT4 Correct)**:

    ```markdown
    GPT4 Correct User: Hello<|end_of_turn|>
    GPT4 Correct Assistant: Hi<|end_of_turn|>
    GPT4 Correct User: How are you today?<|end_of_turn|>
    GPT4 Correct Assistant:
    ```

- **Mathematical Reasoning Mode**:

    ```markdown
    Math Correct User: 10.3 − 7988.8133=<|end_of_turn|>
    Math Correct Assistant:
    ```

    **NOTE**: Remember to set `<|end_of_turn|>` as end of generation token.

- **Integrated Tokenizer**: The default (GPT4 Correct) template is also available as the integrated tokenizer.chat\_template, which can be used instead of manually specifying the template.

#### Suggested labels
#### {   "label": "chat-templates",   "description": "Pre-defined conversation structures for specific modes of interaction." }</details>


### #305: Home - LibreChat
<details><summary>### Details</summary>Similarity score: 0.89
- [ ] [Home - LibreChat](https://docs.librechat.ai/index.html)


Table of contents
🪶 Features
📃 All-In-One AI Conversations with LibreChat
⭐ Star History
✨ Contributors
💖 This project exists in its current state thanks to all the people who contribute
 

LibreChat

       

🪶 Features

🖥️ UI matching ChatGPT, including Dark mode, Streaming, and 11-2023 updates
💬 Multimodal Chat:
Upload and analyze images with GPT-4 and Gemini Vision 📸
More filetypes and Assistants API integration in Active Development 🚧
🌎 Multilingual UI:
English, 中文, Deutsch, Español, Français, Italiano, Polski, Português Brasileiro, Русский
日本語, Svenska, 한국어, Tiếng Việt, 繁體中文, العربية, Türkçe, Nederlands
🤖 AI model selection: OpenAI API, Azure, BingAI, ChatGPT, Google Vertex AI, Anthropic (Claude), Plugins
💾 Create, Save, & Share Custom Presets
🔄 Edit, Resubmit, and Continue messages with conversation branching
📤 Export conversations as screenshots, markdown, text, json.
🔍 Search all messages/conversations
🔌 Plugins, including web access, image generation with DALL-E-3 and more
👥 Multi-User, Secure Authentication with Moderation and Token spend tools
⚙️ Configure Proxy, Reverse Proxy, Docker, many Deployment options, and completely Open-Source
📃 All-In-One AI Conversations with LibreChat

LibreChat brings together the future of assistant AIs with the revolutionary technology of OpenAI's ChatGPT. Celebrating the original styling, LibreChat gives you the ability to integrate multiple AI models. It also integrates and enhances original client features such as conversation and message search, prompt templates and plugins.

With LibreChat, you no longer need to opt for ChatGPT Plus and can instead use free or pay-per-call APIs. We welcome contributions, cloning, and forking to enhance the capabilities of this advanced chatbot platform.

#### Suggested labels
#### "ai-platform"</details>


### #459: llama2
<details><summary>### Details</summary>Similarity score: 0.89
- [ ] [llama2](https://ollama.ai/library/llama2)

Llama 2
========

The most popular model for general use.

*265.8K Pulls*
*Updated 4 weeks ago*

Overview
--------

Llama 2 is released by Meta Platforms, Inc. This model is trained on 2 trillion tokens, and by default supports a context length of 4096. Llama 2 Chat models are fine-tuned on over 1 million human annotations, and are made for chat.

CLI
---

Open the terminal and run

```bash
ollama run llama2
```

API
---

Example using curl:

```bash
curl -X POST http://localhost:11434/api/generate -d '{
  "model": "llama2",
  "prompt":"Why is the sky blue?"
 }'
```

API documentation
-----------------

Memory requirements
-------------------

-  7b models generally require at least 8GB of RAM
-  13b models generally require at least 16GB of RAM
-  70b models generally require at least 64GB of RAM

If you run into issues with higher quantization levels, try using the q4 model or shut down any other programs that are using a lot of memory.

Model variants
--------------

-   **Chat**: fine-tuned for chat/dialogue use cases. These are the default in Ollama, and for models tagged with `-chat` in the tags tab.

    Example: `ollama run llama2`

-   **Pre-trained**: without the chat fine-tuning. This is tagged as `-text` in the tags tab.

    Example: `ollama run llama2:text`

By default, Ollama uses 4-bit quantization. To try other quantization levels, please use the other tags. The number after the `q` represents the number of bits used for quantization (i.e. `q4` means 4-bit quantization). The higher the number, the more accurate the model is, but the slower it runs, and the more memory it requires.

References
----------

-   [Llama 2: Open Foundation and Fine-Tuned Chat Models](https://metastring.com/llama2)
-   [Meta’s Hugging Face repo](https://huggingface.co/Meta)

#### Suggested labels
#### {   "label-name": "llama2-model",   "description": "A powerful text model for chat, dialogue, and general use.",   "repo": "ollama.ai/library/llama2",   "confidence": 91.74 }</details>


### #327: cheshire-cat-ai/core: Production ready AI assistant framework
<details><summary>### Details</summary>Similarity score: 0.88
> **Production ready AI assistant framework**

The Cheshire Cat is a framework to build custom AIs on top of any language model. If you have ever used systems like WordPress or Django to build web apps, imagine the Cat as a similar tool, but specific for AI.

**Quickstart**

To make Cheshire Cat run on your machine, you just need docker installed:

```
docker run --rm -it -p 1865:80 ghcr.io/cheshire-cat-ai/core:latest
```

Chat with the Cheshire Cat on localhost:1865/admin.

You can also interact via REST API and try out the endpoints on localhost:1865/docs

As a first thing, the Cat will ask you to configure your favourite language model. It can be done directly via the interface in the Settings page (top right in the admin).

Enjoy the Cat!

Follow instructions on how to run it with docker compose and volumes.

**Docs and Resources**

- [Official Documentation](https://github.com/cheshire-cat-ai/core)
- [Discord Server](https://github.com/cheshire-cat-ai/core)
- [Website](https://github.com/cheshire-cat-ai/core)
- [YouTube tutorial - How to install](https://github.com/cheshire-cat-ai/core)
- [Tutorial - Write your first plugin](https://github.com/cheshire-cat-ai/core)

**Why use the Cat**

- Supports any language model (works with OpenAI chatGPT, Llama2, HuggingFace models, custom)
- Remembers conversations and documents and uses them in conversation
- Extensible via plugins (AI can connect to your APIs or execute custom python code)
- Production ready - 100% dockerized
- Active Discord community and easy to understand docs

We are committed to openness, privacy and creativity, we want to bring AI to the long tail. If you want to know more about our vision and values, read the Code of Ethics.

**Roadmap**

Detailed roadmap is [here](https://github.com/cheshire-cat-ai/core). Whilst for the current progress of development, take a look at the projects marked as open.

**Contributing**

To get the full dev setup, follow install instructions. Send your pull request to the develop branch. Here is a full guide to contributing.

Join our community on Discord and give the project a star ⭐! Thanks again!🙏

> **Which way to go?**

> "Would you tell me, please, which way I ought to go from here?"

> "That depends a good deal on where you want to get to," said the Cat.

> "I don't much care where--" said Alice.

> "Then it doesn't matter which way you go," said the Cat.

> (Alice's Adventures in Wonderland - Lewis Carroll)

#### Suggested labels
- { "key": "AI-Chatbots", "value": "Topics related to advanced chatbot platforms integrating multiple AI models" }
- { "key": "llama", "value": "llama models and tools" }</details>


### #485: Docs | OpenRouter
<details><summary>### Details</summary>Similarity score: 0.88
- [ ] [Docs | OpenRouter](https://openrouter.ai/docs#models)

 # Title: Docs | OpenRouter

Description: The future will bring us hundreds of language models and dozens of providers for each. How will you choose the best?

Benefit from the race to the bottom. OpenRouter finds the lowest price for each model across dozens of providers. You can also let users pay for their own models via OAuth PKCE.

Standardized API. No need to change your code when switching between models or providers.

The best models will be used the most. Evals are flawed. Instead, compare models by how often they're used, and soon, for which purposes. Chat with multiple at once in the Playground.

URL: <https://openrouter.ai/docs#models>

## Key Features

- **Lowest Price Guarantee:** OpenRouter finds the lowest price for each model across dozens of providers.
- **Standardized API:** No need to change your code when switching between models or providers.
- **Usage-Based Comparison:** Compare models by how often they're used, and soon, for which purposes.
- **User-Paid Models:** Allow users to pay for their own models via OAuth PKCE.
- **Playground:** Chat with multiple models at once in the Playground.

OpenRouter is the future of language model selection and usage. Benefit from a wide range of models and providers, while ensuring the best models are used the most.

#### Suggested labels
#### {   "label-name": "language-models",   "description": "Information about language models and providers",   "repo": "openrouter.ai",   "confidence": 96.2 }</details>


### #552: LargeWorldModel/LWM-Text-Chat-1M · Hugging Face
<details><summary>### Details</summary>Similarity score: 0.87
- [ ] [LargeWorldModel/LWM-Text-Chat-1M · Hugging Face](https://huggingface.co/LargeWorldModel/LWM-Text-Chat-1M)


# LargeWorldModel/LWM-Text-Chat-1M · Hugging Face

**DESCRIPTION:**

LWM-Text-1M-Chat Model Card

**Model details**

Model type: LWM-Text-1M-Chat is an open-source model trained from LLaMA-2 on a subset of Books3 filtered data. It is an auto-regressive language model, based on the transformer architecture.

Model date: LWM-Text-1M-Chat was trained in December 2023.

Paper or resources for more information: [https://largeworldmodel.github.io/](https://largeworldmodel.github.io/)

**URL:** [https://huggingface.co/LargeWorldModel/LWM-Text-Chat-1M](https://huggingface.co/LargeWorldModel/LWM-Text-Chat-1M)

#### Suggested labels
#### {'label-name': 'Open-source Models', 'label-description': 'Models that are publicly available and open-source for usage and exploration.', 'gh-repo': 'huggingfaceco/LargeWorldModel/LWM-Text-Chat-1M', 'confidence': 56.11}</details>

**Comment by williamewing9087** [comment]
nice
